streamer/src/recvmmsg.rs

```diff
@@ -126,11 +142,24 @@ pub fn recv_mmsg(sock: &UdpSocket, packets: &mut [Packet]) -> io::Result</*num p
         usize::try_from(nrecv).unwrap()
     };
     for (addr, hdr, pkt) in izip!(addrs, hdrs, packets.iter_mut()).take(nrecv) {
-        pkt.meta_mut().size = hdr.msg_len as usize;
-        if let Some(addr) = cast_socket_addr(&addr, &hdr) {
+        let hdr_ref = unsafe { hdr.assume_init_ref() };
+        let addr_ref = unsafe { addr.assume_init_ref() };
+        pkt.meta_mut().size = hdr_ref.msg_len as usize;
+        if let Some(addr) = cast_socket_addr(addr_ref, hdr_ref) {
             pkt.meta_mut().set_socket_addr(&addr);
         }
     }
+
+    // TODO - figure out in review whether we think we have to drop these OR if
+    //        leaving as MaybeUninit (no drop) is ok
+    for (iov, addr, hdr) in izip!(&mut iovs, &mut addrs, &mut hdrs).take(count) {
```

Comment:
i think this should be count.min(nrecv)?

Reply:
I think `count` may still be correct:
- We initialize `count` buffers for `recvmmsg()` to populate
- `recvmmsg()` populates `nrecv` buffers, where `0 <= nrecv <= count`

So, in the event that `nrecv < count`, only the first `nrecv` items of `iovs` and `addrs` are filled with payload and src IP. However, the other `count - nrecv` items were still initialized properly, they just didn't get any data pushed in by `recvmmsg()`

Reply:
ah yep of course you're right!